include/zephyr/usb/usbd_msg.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief USB support message types and structure
+ */
+
+#ifndef ZEPHYR_INCLUDE_USBD_MSG_H_
+#define ZEPHYR_INCLUDE_USBD_MSG_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup usbd_msg_api USB device core API
+ * @ingroup usb
+ * @{
+ */
+
+/**
+ * @brief USB device support message types
+ *
+ * The first set of message types map to event types from the UDC driver API,
+ * reserved types are not used.
+ */
+enum usbd_msg_type {
+	/** VBUS ready message (optional) */
+	USBD_MSG_VBUS_READY,
+	/** VBUS removed message (optional) */
+	USBD_MSG_VBUS_REMOVED,
+	/** Device resume message */
+	USBD_MSG_RESUME,
+	/** Device suspended message */
+	USBD_MSG_SUSPEND,
+	/** Bus reset detected */
+	USBD_MSG_RESET,
+	/** Reserved */
+	USBD_MSG_RESERVED0,
+	/** Reserved */
+	USBD_MSG_RESERVED1,
```

Comment:
What's the point of having reserved values here? Isn't just keeping API compatibility (i.e. not renaming the entries, but reordering is fine) enough here?